src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncExceptionHandlerRewriter.cs

```diff
@@ -141,7 +141,8 @@ public override BoundNode VisitTryStatement(BoundTryStatement node)
                 tryStatementSyntax.IsKind(SyntaxKind.UsingStatement) ||
                 tryStatementSyntax.IsKind(SyntaxKind.ForEachStatement) ||
                 tryStatementSyntax.IsKind(SyntaxKind.ForEachVariableStatement) ||
-                tryStatementSyntax.IsKind(SyntaxKind.LocalDeclarationStatement));
+                tryStatementSyntax.IsKind(SyntaxKind.LocalDeclarationStatement) ||
+                tryStatementSyntax.IsKind(SyntaxKind.LockStatement));
```

Comment:
:memo: I validated that adding/removing a lock statement in an async method with EnC works.